lib/pg_party/adapter_decorator.rb

```diff
@@ -26,17 +33,42 @@ def create_list_partition_of(table_name, values:, **options)
       create_partition_of(table_name, list_constraint_clause(values), **options)
     end
 
+    def create_hash_partition_of(table_name, modulus:, remainder:, **options)
+      create_partition_of(table_name, hash_constraint_clause(modulus, remainder), **options)
+    end
+
+    def create_default_partition_of(table_name, **options)
+      create_partition_of(table_name, nil, default_partition: true, **options)
+    end
+
     def create_table_like(table_name, new_table_name, **options)
-      primary_key = options.fetch(:primary_key) { calculate_primary_key(table_name) }
+      primary_key           = options.fetch(:primary_key) { calculate_primary_key(table_name) }
+      partition_key         = options.fetch(:partition_key, nil)
+      partition_type        = options.fetch(:partition_type, nil)
+      create_with_pks       = options.fetch(
+                                :create_with_primary_key,
+                                PgParty.config.create_with_primary_key
+                              )
+
+      validate_primary_key(primary_key) unless create_with_pks
+      if partition_type
+        validate_supported_partition_type!(partition_type)
+        raise ArgumentError, '`partition_key` is required when specifying a partition_type' unless partition_key
+      end
 
-      validate_primary_key(primary_key)
+      like_option = if !partition_type || create_with_pks
+                      'INCLUDING ALL'
+                    else
+                      'INCLUDING ALL EXCLUDING INDEXES'
```

Comment:
If we are excluding indexes when creating template tables, do template tables have any uses at this point? Should the create with PKs and template table config options be mutually exclusive?

Reply:
I think they do, 1) for preserving backward compatibility for folks already using them, and 2) because if you have expressions in your partition key you cannot use postgres 11/12's primary key support on partitioned tables.

Template tables are more flexible and I figure people may already be using them in Postgres 11+ but may want to use the native features on new tables and mix the features to get the best of both worlds... could be revisited in a version 2 of the gem...